### PR TITLE
Enable threadpool in evaluate API

### PR DESCRIPTION
**Pull request by ninghu**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by nagkumar91 — APPROVED** [review]

**Review by singankit — COMMENTED** [review]
We should not need this here. Code client internally already does parallelization on each row in of data for an evaluator and on each evaluator

**Review by nagkumar91 — APPROVED** [review]

**Review by singankit — COMMENTED** [review]
Should we remove this part since this is covered in code client anyway ?

**Review by nick863 — DISMISSED** [review]

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 1]
```diff
@@ -372,31 +321,37 @@ def evaluate(
...
     for evaluator_name, evaluator in evaluators.items():
-        evaluator_info[evaluator_name] = {}
-        evaluator_info[evaluator_name]["run"] = pf_client.run(
+        evaluator_info[evaluator_name] = {
+            "client": batch_run_client,
+        }
+        evaluator_info[evaluator_name]["run"] = batch_run_client.run(
             flow=evaluator,
             run=target_run,
+            evaluator_name=evaluator_name,
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:334]: Does this param exist on pf_client ?

**Review by luigiw — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py (hunk 1 above)
Comment (ninghu) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:333]: Checked. It doesn't exist on pf_client.

**Review by singankit — APPROVED** [review]

**Pull request merged**